lib/PhantomEnvironment.js

```diff
@@ -53,6 +54,69 @@ const _ = require('lodash'),
  */
 
 /**
+ * @param {ProxyIndicator} proxyIndicator
+ * @returns {Error}
+ */
+function createProxyError(proxyIndicator) {
+  let msg;
+  switch (proxyIndicator.type) {
+    case 'redirect':
+      msg = 'Proxy matched redirect';
+      break;
+    case 'responseCode':
+      msg = 'Proxy matched response code';
+      break;
+    case 'captcha':
+      msg = 'Captcha handled';
+      break;
+    default:
+      throw new Error('Unsupported proxyIndicator');
+  }
+  const err = new Error(msg);
+  err.proxyIndicator = proxyIndicator.type;
+  err.proxyLevel = proxyIndicator.level || 'medium';
+
+  return err;
+}
+
+/**
+ * @param {string} currentUrl
+ * @param {string} redirectUri
+ * @returns {string}
+ * @private
+ */
+function getRedirectUrl(currentUrl, redirectUri) {
```

Comment:
Why second param ends with `i` and first with `l`. Shall it be the same `Url` ending?

Reply:
signature was not changed, PR is about promises at most. It initially was written by you, so the question to you lol 🐗